src/components/MoneyReportHeader.js

```diff
@@ -52,13 +55,22 @@ const defaultProps = {
 };
 
 function MoneyReportHeader(props) {
+    const {translate} = useLocalize();
     const moneyRequestReport = props.report;
+    const policy = props.policies[`${ONYXKEYS.COLLECTION.POLICY}${moneyRequestReport.policyID}`];
+    const isPolicyAdmin = policy.role === CONST.POLICY.ROLE.ADMIN;
+    const isManager = ReportUtils.isMoneyRequestReport(moneyRequestReport) && lodashGet(props.session, 'accountID', null) === moneyRequestReport.managerID;
+    const isReportApproved = ReportUtils.isReportApproved(moneyRequestReport);
+    const isPayer = CONST.POLICY.TYPE.CORPORATE ? isPolicyAdmin && isReportApproved : isPolicyAdmin || (ReportUtils.isMoneyRequestReport(moneyRequestReport) && isManager);
     const isSettled = ReportUtils.isSettled(moneyRequestReport.reportID);
-    const policy = props.policies[`${ONYXKEYS.COLLECTION.POLICY}${props.report.policyID}`];
-    const isPayer =
-        Policy.isAdminOfFreePolicy([policy]) || (ReportUtils.isMoneyRequestReport(moneyRequestReport) && lodashGet(props.session, 'accountID', null) === moneyRequestReport.managerID);
+    const shouldShowSettlementButton = useMemo(() => isPayer && !isSettled, [isPayer, isSettled]);
+    const shouldShowApproveButton = useMemo(() => {
+        if (policy.type === CONST.POLICY.TYPE.FREE) {
+            return false;
+        }
+        return isManager && isReportApproved && !isSettled;
```

Comment:
It should not be approved yet, right?
```suggestion
        return isManager && !isReportApproved && !isSettled;
```